server/src/main/java/io/spine/server/entity/TransactionalEntity.java

```diff
@@ -160,40 +167,58 @@ final boolean isTransactionInProgress() {
 
     @SuppressWarnings({"ObjectEquality", "ReferenceEquality"}
             /* The refs must to point to the same object; see below. */)
-    void injectTransaction(Transaction<I, ? extends TransactionalEntity<I, S, B>, S, B> tx) {
+    final void injectTransaction(Transaction<I, ? extends TransactionalEntity<I, S, B>, S, B> tx) {
         checkNotNull(tx);
-
         /*
             To ensure we are not hijacked, we must be sure that the transaction
             is injected to the very same object and wrapped into the transaction.
         */
         checkState(tx.entity() == this,
-                   "Transaction injected to this " + this
-                           + " is wrapped around a different entity: " + tx.entity());
+                   "Transaction injected to this %s" +
+                           " is wrapped around a different entity: %s.", this, tx.entity());
 
         this.transaction = tx;
     }
 
-    void releaseTransaction() {
+    final void releaseTransaction() {
         this.transaction = null;
     }
 
-    @Nullable
+    /**
+     * Obtains the transaction which modifies this entity.
+     *
+     * <p>This is a test-only method. For production purposes please use {@link #tx()}.
+     *
+     * @return the instance of the transaction or {@code null} if the entity is not being modified
+     * @see #tx()
+     */
     @VisibleForTesting
-    Transaction<I, ? extends TransactionalEntity<I, S, B>, S, B> getTransaction() {
+    final @Nullable Transaction<I, ? extends TransactionalEntity<I, S, B>, S, B> transaction() {
         return transaction;
     }
 
     /**
      * Updates own {@code stateChanged} flag from the underlying transaction.
      */
-    void updateStateChanged() {
+    final void updateStateChanged() {
         this.stateChanged = tx().isStateChanged();
     }
 
-    B builderFromState() {
+    /**
+     * Obtains the builder for being used by a transaction.
```

Comment:
This seems too specific to me. Do we assume it is only used in transactions?

Reply:
It is used only in the `Transaction` and tests related to it.

Reply:
Can it be used elsewhere?
Some methods of `TransactionalEntity` may only be used while in a transaction and throw exceptions otherwise. Is it the case with this method?

Reply:
Moved to `Transaction`.